src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,208 @@
+        Node<T> newNode = new Node(value);
+        if (index == 0 && listCount == 0) {
+            add(value);
+        } else if (index == 0 && listCount > 0) {
+            newNode.next = first;
+            newNode.prev = null;
+            first.prev = newNode;
+            first = newNode;
+            listCount++;
+        } else if (index == listCount) {
+            add(value);
+        } else {
+            Node<T> oldNode = first;
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            addItem(oldNode, newNode);
+            listCount++;
+
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> node = first;
+        if (index == 0) {
+            return (T) first.value;
+        }
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return (T) node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> oldNode = first;
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            newNode.next = first.next;
+            newNode.prev = first.prev;
+            first.next.prev = newNode;
+            first = newNode;
+        } else if (index == listCount - 1) {
+            oldNode = last;
+            newNode.next = null;
+            newNode.prev = last.prev;
+            last.prev.next = newNode;
+            newNode = last;
+        } else {
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            addItem(oldNode, newNode);
+        }
+        return (T) oldNode.value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> oldNode = first;
+        if (index == 0 && size() == 1) {
+            first = null;
+            last = null;
+        } else if (index == 0) {
+            first.next.prev = null;
+            first = first.next;
+        } else if (index == listCount - 1) {
+            oldNode = last;
+            last.prev.next = null;
+            last.prev = last;
+            last.prev = null;
+        } else {
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            oldNode.prev.next = oldNode.next;
+            oldNode.next.prev = oldNode.prev;
+        }
+        listCount--;
+        return oldNode.value;
```

Comment:
too complex logic, let's first find a node that will be removed, after that I guess you could see the ways of simplifying of your logic